theano/sandbox/cuda/dnn.py

```diff
@@ -443,8 +443,8 @@ def grad(self, inp, grads):
 
         top = gpu_contiguous(top)
 
-        d_img = GpuDnnConvGradI()(kerns, top, img.zeros_like(), desc)
-        d_kerns = GpuDnnConvGradW()(img, top, kerns.zeros_like(), desc)
+        d_img = GpuDnnConvGradI()(kerns, top, img, desc)
+        d_kerns = GpuDnnConvGradW()(img, top, kerns, desc)
```

Comment:
But then it won't be able to reuse the allocated `img` from the forward pass in the backward pass, would it (in the case of training a CNN)? Is there a way to make that happen? It would probably require the inplace optimizations to run in a particular order to end up with this solution, but it could allow larger networks to be trained. (Or couldn't it? I'm not really familiar with how Theano manages memory. Will it run `GpuDnnConvGradW` first and then free `img` before running `GpuDnnConvGradI`, if `allow_gc=1`?)

In any case, I'm fine with `GpuAllocEmpty` as I don't use `dnn_conv()` directly anyway, but I'd be interested in hearing your thoughts.